### PR TITLE
Add a label which prevents a node from being added to a cloud load ba…

### PR DESCRIPTION
**Pull request by rbitia**
…lancer.

**What this PR does / why we need it**:
Cherry picked from the master/kubernetes branch with this commit #53146 and merged it with release 1.8 so we can get the aci-connector issue fixed which is here: https://github.com/Azure/aci-connector-k8s/issues/51 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

https://github.com/Azure/aci-connector-k8s/issues/51

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ const (
 	// It's copied over to kubeadm until it's merged in core: https://github.com/kubernetes/kubernetes/pull/39112
 	LabelNodeRoleMaster = "node-role.kubernetes.io/master"
 
+	// LabelNodeRoleExcludeBalancer specifies that the node should be
+	// exclude from load balancers created by a cloud provider.
+	LabelNodeRoleExcludeBalancer = "node.role.kubernetes.io/exclude-balancer"
```
Comment (smarterclayton) [inline cmd/kubeadm/app/constants/constants.go]: Is there a reason why this label is `node.role` and the one above is `node-role`?

Comment (jdumars) [reply cmd/kubeadm/app/constants/constants.go]: @brendandburns

File: pkg/controller/service/service_controller.go [hunk 2]
```diff
@@ -598,6 +598,10 @@ func getNodeConditionPredicate() corelisters.NodeConditionPredicate {
 			return false
 		}
 
+		if _, hasExcludeBalancerLabel := node.Labels[constants.LabelNodeRoleExcludeBalancer]; hasExcludeBalancerLabel {
```
Comment (mikedanese) [inline pkg/controller/service/service_controller.go]: I just opened https://github.com/kubernetes/kubernetes/issues/53341. We shouldn't depend on cmd/kubeadm from core controllers.

**Comment by mikedanese** [comment]
@kubernetes/sig-network-api-reviews

**Comment by mikedanese** [comment]
This is a cherrypick.

**Comment by bowei** [comment]
@mikedanese -- is this really a cherrypick, as I see some comments about the approach taken...

**Comment by mikedanese** [comment]
@bowei the original got in without much review.  #53146

**Comment by jdumars** [comment]
@mikedanese @smarterclayton how do we best resolve the underlying concerns of the original merge, and does that make this cherrypick too risky to put in 1.8.1?

**Comment by smarterclayton** [comment]
We probably need to resolve the naming question in the original PR - we don't want to backport something inconsistent.

**Comment by mikedanese** [comment]
Fixing the dependency is not an issue for 1.8 but I agree we should fix the . vs - before cherrypicking. Also an ack from a sig-networking api reviewer would be nice since this is an api change.

**Comment by jdumars** [comment]
Obviated by #53656 
@rbitia please close this one

**Pull request closed without merging**